icu4j/main/core/src/main/java/com/ibm/icu/impl/JavaTimeConverters.java

```diff
@@ -0,0 +1,306 @@
+// © 2024 and later: Unicode, Inc. and others.
+// License & terms of use: https://www.unicode.org/copyright.html
+
+package com.ibm.icu.impl;
+
+import static java.time.ZoneOffset.UTC;
+import static java.time.temporal.ChronoField.MILLI_OF_SECOND;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.chrono.ChronoLocalDate;
+import java.time.chrono.ChronoLocalDateTime;
+import java.time.temporal.Temporal;
+import java.util.Date;
+
+import com.ibm.icu.util.Calendar;
+import com.ibm.icu.util.GregorianCalendar;
+import com.ibm.icu.util.SimpleTimeZone;
+import com.ibm.icu.util.TimeZone;
+import com.ibm.icu.util.ULocale;
+
+
+/**
+ * This class provides utility methods for converting between Java 8's {@code java.time}
+ * classes and the {@link com.ibm.icu.util.Calendar} and related classes from the
+ * {@code com.ibm.icu.util} package.
+ *
+ * <p>
+ * The class includes methods for converting various temporal types, such as
+ * {@link Instant}, {@link ZonedDateTime}, {@link OffsetTime}, {@link OffsetDateTime}, {@link LocalTime},
+ * {@link ChronoLocalDate}, and {@link ChronoLocalDateTime}, to {@link Calendar} instances.
+ *
+ * <p>
+ * Additionally, it provides methods to convert between {@link ZoneId} and {@link TimeZone}, and
+ * {@link ZoneOffset} and {@link TimeZone}.
+ *
+ * @deprecated This API is ICU internal only.
+ */
+@Deprecated
+public class JavaTimeConverters {
+    // Milliseconds per day
+    private static final long MILLIS_PER_DAY = 24 * 60 * 60 * 1_000;
+
+    /**
+     * Converts the current instant from a {@link Clock} to a {@link Calendar}.
+     *
+     * <p>
+     * This method creates a {@link Calendar} instance that represents the current
+     * instant as provided by the specified {@link Clock}. The resulting
+     * {@link Calendar} will be in the default time zone of the JVM.
```

Comment:
question: Why does this comment say that the `Calendar` will be in the default time zone of the JVM? Isn't the code using the time zone of the provided `Clock`?

Reply:
Bad comment, copy-pasted from `Instant` (which does not have a time zone).

I removed the extra sentence.